sdcm/tester.py

```diff
@@ -782,19 +783,39 @@ def init_resources(self, loader_info=None, db_info=None,
         elif cluster_backend == 'baremetal':
             self.get_cluster_baremetal()
 
+    def set_seeds(self):
+        seeds_selector = self.params.get('seeds_selector', default='first')
+        assert seeds_selector in ['reflector', 'random', 'first'], 'Unknown seed selector. ' \
+                                                                   'Expected values: reflector / random / first'
         seeds_num = self.params.get('seeds_num', default=1)
-        if self.params.get('instance_provision') == 'mixed':
-            for node in self.db_cluster.nodes:
-                if seeds_num == 0:
-                    break
-                if node.is_spot:
-                    node.is_seed = True
-                    seeds_num -= 1
+        assert seeds_num > 0, 'Seed number can\'t be zero'
+
+        if seeds_selector == 'reflector':
+            node = self.db_cluster.nodes[0]
+            node.wait_ssh_up()
+            self.db_cluster.seed_nodes_ips = node.extract_seeds_from_scylla_yaml()
         else:
-            for i in range(seeds_num):
-                self.db_cluster.nodes[i].is_seed = True
-                if self.cs_db_cluster:
-                    self.cs_db_cluster.nodes[i].is_seed = True
+            assert seeds_num <= len(self.db_cluster.nodes), \
+                'Seeds number ({}) should be less then nodes number ({})'.format(seeds_num, len(self.db_cluster.nodes))
+
+            instance_provision = self.params.get('instance_provision')
+            selected_seeds = set()
+            cnt = 0
+            while len(selected_seeds) < seeds_num and cnt <= len(self.db_cluster.nodes):
```

Comment:
Too complicated...
- random seed selector: use `random.sample`
- first just use  self.db_cluster.nodes[:num_seeds]

Reply:
I did it on such way because of in case mixed instance_provision, we need additional validation (below)

Reply:
Bentsi: in case "mixed"  instance_provision, let warning

Reply:
done